src/main/java/com/compilercharisma/chameleonbusinessstudio/service/AppointmentService.java

```diff
@@ -56,6 +58,83 @@ private Mono<Appointment> addAppointmentToParticipants(Appointment appt){
             .then(Mono.just(appt));
     }
 
+    /**
+     * Creates many appointments at once. This method is prefered to multiple
+     * calls to createAppointment, as the latter makes more requests and is
+     * vulnerable to race conditions.
+     * 
+     * @param appts the appointments to create
+     * @return a flux containing the created appointments
+     */
+    public Flux<Appointment> createAppointments(List<Appointment> appts){
+        /*
+         * Since addAppointmentForUser uses Update in Vendia to update a user’s 
+         * appointment array, the following event may occur:
+         * 
+         * Suppose there exists a user in Vendia with the following values:
+         *  email = user1; appointments = [appt0]
+         * Suppose 
+         *  addAppointmentForUser(appt1, user1) 
+         * and 
+         *  addAppointmentForUser(appt2, user1) 
+         * are called in parallel within a Flux.
+         * Suppose addAppointmentForUser(appt1, user1) begins resolving, and 
+         * retrieves the user from Vendia. It then correctly updates their 
+         * appointments to [appt0, appt1]. Then, it makes a call to post the 
+         * updates user to Vendia.
+         * However, there is nothing preventing 
+         * addAppointmentForUser(appt2, user1) from starting before Vendia is 
+         * updated. Therefore, it will get the user from Vendia, see their 
+         * appointments as [appt0], then update it to [appt0, appt2]. It then 
+         * makes the call to post the updated user to Vendia. Because of this, 
+         * only one of the appointments has been correctly added to the user in 
+         * Vendia.
+         * 
+         * This method was written specifically to solve that problem.
+         */
```

Comment:
Great documentation, really helped me understand the nature of this potential race condition.